### PR TITLE
rebase: update github.com/libopenstorage/secrets to latest

### LINKED ISSUE #4196: Update libopenstorage/secrets to latest version
https://github.com/ceph/ceph-csi/blob/5ff0607360815cd2c9ba8a420ceacbd2ce298ff5/go.mod#L24C2-L24C70

Update the libopenstorage/secrets to the latest version to remove `github.com/hashicorp/vault` dependency in `go.mod`. 

More info: https://github.com/libopenstorage/secrets/pull/80

### PR DESCRIPTION
**Pull request by Rakshith-R**
With this update, we no longer import github.com/hashicorp/vault which now is under BSL license.
https://github.com/hashicorp/vault/blob/main/LICENSE

resolves: #4196

### CODE REVIEW

**Review by Madhu-1 — DISMISSED** [review]

**Comment by Madhu-1** [comment]
@Rakshith-R ci failure please check

**Comment by Madhu-1** [comment]
@Rakshith-R jfyi we use pretty old version of vault, currently we are not affected by this

**Comment by Rakshith-R** [comment]
> @Rakshith-R jfyi we use pretty old version of vault, currently we are not affected by this

Yes, but its still good to remove any traces of the main repo to avoid any confusion later on ?

File: internal/kms/vault.go [hunk 1]
```diff
@@ -422,7 +422,7 @@ func (kms *vaultKMS) StoreDEK(key, value string) error {
 	}
 
 	pathKey := filepath.Join(kms.vaultPassphrasePath, key)
-	err := kms.secrets.PutSecret(pathKey, data, kms.keyContext)
+	_, err := kms.secrets.PutSecret(pathKey, data, kms.keyContext)
```
Comment (Madhu-1) [inline internal/kms/vault.go:427]: add a comment on what exactly is getting discarded here. same in other places as well.

**Review by Madhu-1 — COMMENTED** [review]
LGTM, left a suggestion

**Comment by Madhu-1** [comment]
> > @Rakshith-R jfyi we use pretty old version of vault, currently we are not affected by this
> 
> Yes, but its still good to remove any traces of the main repo to avoid any confusion later on ?

Yes that's a good idea but i don't prefer to backport this to the release branch

**Review by Madhu-1 — APPROVED** [review]

**Review by riya-singhal31 — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio queue

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e/k8s-1.26

Failed with:
```
[FAILED] failed to execute rados command 'rados listomapkeys csi.volumes.default --pool=replicapool | wc -l' : stdErr=2023-10-17T08:37:35.457+0000 7f7ef5ff3700 -1 monclient: handle_auth_reply_more no authorizer?
```

[logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.26/detail/mini-e2e_k8s-1.26/633/pipeline/)

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**